.travis.yml

```diff
@@ -0,0 +1,19 @@
+sudo: false
+
+services:
+  - docker
+
+before_script:
+  - docker -v
+  - docker build -t keboola/ex-http .
```

Comment:
Nechal jsem si to přes noc projít hlavou a vidím tady v použití čistého dockeru ještě jeden potenciální problém. A to, že **pokud jsou někde na build potřeba další servicy** (třeba databáze, redis nebo tak něco), **tak se to bude muset dělat buď ručně  nebo přepsat zpátky na compose**. 

Zkusil jsem si okolo toho něco načíst, ale vypadá to, že to fakt nikdo nijak "standardizovaně" neřeší.

Hlavní problém, který řešíme je, že bysme chtěli mít jakoby víc konfigurací najednou (jednu na vývoj, jednu na travis, jednu na produkci). Mělo by jít [použít víc docker-compose souborů a ty pozdějí overridují ten dřívější](https://docs.docker.com/compose/extends/#multiple-compose-files). Ale znamenalo by to mít v rootu jen tu defaultní "produkční" konfiguraci. A pak mít ještě šložku s travis konfigurací. A volat to `docker-compose -f docker-compose.yml -f docker-compose/dev.yml up` resp. `docker-compose -f docker-compose.yml -f docker-compose/travis.yml up`. Případně mít ten root `docker-compose.yml` pro vývojáře (dev) a ve složce mít extra compose pro build a produkci (tam parametr navíc nevadí - je to stejně napevno napsané v buildu). 
Každopádně to zvyšuje složitost. 

Nicméně pomocí tohohle a [extension points](https://docs.docker.com/compose/compose-file/#extension-fields) by to šlo udělat DRY. Ale stále bysme se nevyhnuli více souborům :( 

**Je ten scénář s více službami potřebnými k testování reálný?** Mohl bych projít všechny repozitáře a podívat se, ale přeci jen asi bude rychlejší, když mi to někdo kdo má přehled řekne :)

Reply:
Neni realny, vetsina komponenta potrebuje jen samu sebe, A kdyz ne, tak bych to resil az to bude aktualni, anebo proste nejak jinak - treba generic ma custom testy https://github.com/keboola/generic-extractor/blob/master/.travis.yml protoze si jeste testuje example a proxy a tech vic konfiguraci tam zadnou neplechu nedela. Asi by to slo resit pres vic docker-compose souboru, ale takhle me to prijde v pohode

Reply:
Mohl bys třeba pro mysql extraktor chtít pustit mysql server lokálně v dockeru, ale asi nevidím problém pustit `docker-compose up mysql` lokálně i na travisu a pak to prolinkovat s testovaným kontejnerem přes `docker run --link...`?

Reply:
Super, jestli jste s tím OK, tak fajn.